components/intercom/intercom.app.mjs

```diff
@@ -0,0 +1,232 @@
+
+export default {
+  type: "app",
+  app: "intercom",
+  propDefinitions: {
+    userIds: {
+      type: "string[]",
+      label: "Users",
+      description: "Users to watch for new events",
+      async options() {
+        const data = {
+          query: {
+            field: "role",
+            operator: "=",
+            value: "user",
+          },
+        };
+        const results = await this.searchContacts(data);
+        return results.map((user) => ({
+          label: user.name || user.id,
+          value: user.id,
+        }));
+      },
+    },
+    body: {
+      type: "string",
+      label: "Body",
+      description: "The text of the note.",
+    },
+  },
+  methods: {
+    monthAgo() {
+      const monthAgo = new Date();
+      monthAgo.setMonth(monthAgo.getMonth() - 1);
+      return monthAgo;
+    },
+    /**
+    * Make a request to the Intercom API
+    * @params {Object} opts - An object representing the configuration options for this method
+    * @params {String} opts.method - The HTTP method
+    * @params {String} [opts.url] - A full URL to be used in the API request. If not present,
+    * endpoint is used.
+    * @params {String} [opts.endpoint] - Endpoint to be after the base URL in the API request.
+    * @params {Object} [opts.data] - The request body
+    * @returns {*} The response may vary depending on the specific API request.
+    */
+    async makeRequest(opts) {
+      const {
+        method,
+        url,
+        endpoint,
+        data,
+        $,
+      } = opts;
+      const config = {
+        method,
+        url: url
+          ? url
+          : `https://api.intercom.io/${endpoint}`,
+        headers: {
+          Authorization: `Bearer ${this.$auth.oauth_access_token}`,
+          Accept: "application/json",
+        },
+        data,
+      };
+      return await axios($ || this, config);
+    },
+    /**
+     * Paginate through a list of items and return the results
+     * @params {String} itemType - The type of item being retrieved
+     * (eg. "companies", "contacts", "conversation")
+     * @params {String} method - The HTTP method
+     * @params {Object} data - The request body
+     * @params {Boolean} [isSearch=false] - If set to true, adds the word "search" to the
+     * endpoint for search requests
+     * @params {Interger} [lastCreatedAt] - Timestamp of the last relevant item created.
+     * Used to retrieve only new results
+     * @returns {Array} The complete list of paginated items
+     */
+    async paginate(itemType, method, data, isSearch = false, lastCreatedAt) {
+      let results = null;
+      let done = false;
+      let items = [];
+      while ((!results || results?.pages?.next) && !done) {
+        const startingAfter = results?.pages?.next?.starting_after || null;
+        const endpoint = `${itemType}/${isSearch
+          ? "search"
+          : ""}${startingAfter
+          ? "?starting_after=" + startingAfter
+          : ""}`;
```

Comment:
Optionally. I think these lines are a little hard to understand, Maybe I would recommend doing it like this?
```suggestion
        const search = isSearch && "/search" || "";
        const startingAfterParam = startingAfter && `?starting_after=${startingAfter}` || "";
        const endpoint = `${itemType}${search}${startingAfterParam}`;
```